### PR TITLE
Publish intensity event during scanDurationDt and fix scanDurationDt

### PR DESCRIPTION
**Pull request by couger01**

### CODE REVIEW

File: python/lsst/ts/electrometer/controller.py
Comment (r-owen) [inline python/lsst/ts/electrometer/controller.py:248]: I suggest using lsst.ts.utils.current_tai for all timestamps.

It's perhaps overkill for measuring durations, but values like this often end up being logged or published in SAL topics (even if you don't do that now).

File: python/lsst/ts/electrometer/controller.py
Comment (r-owen) [inline python/lsst/ts/electrometer/controller.py:58]: At line 49 and 51: manual_start_time and manual_end_time are float, not int. And if you change them to TAI (requested below) then please document the units as TAI seconds.

File: python/lsst/ts/electrometer/controller.py
Comment (r-owen) [inline python/lsst/ts/electrometer/controller.py:263]: At line 251 you say "Start storing values to the buffer" and similarly for `start_scan`.

It took me awhile to realize the buffer was in the Keithly. Please add a few more words to that effect to *both* doc strings. For this one it could be as simple as "Start storing values in the Kiethley electrometer, for a set duration".

File: python/lsst/ts/electrometer/controller.py
Comment (r-owen) [inline python/lsst/ts/electrometer/controller.py:381]: I strongly suggest you eliminate `self.last_value` and simply return the value from `get_intensity` because:
* You aren't using it anywhere except to report it here.
* It duplicates the information stored in self.csc.evt_intensity. So if you did want to get the most recently read value you can read it from there. This avoids having two copies that can get out of sync with each other.

Comment (r-owen) [reply python/lsst/ts/electrometer/controller.py:381]: Please add a doc string.

File: python/lsst/ts/electrometer/csc.py
Comment (r-owen) [inline python/lsst/ts/electrometer/csc.py:265]: I gently suggest deleting this method and calling self.evt_detailedState.set_write(detailedState=detailed_state) directly.

I admit that the method does validation, but this is internal and you are already being careful to only write Electrometer.DetailedState values.

File: python/lsst/ts/electrometer/csc.py
Comment (r-owen) [inline python/lsst/ts/electrometer/csc.py:54]: I suggest changing line 2 to:
```
from lsst.ts.idl.enums.Electrometer import DetailedState
```
because to readers of your code `Electrometer` looks like the local `electrometer` namespace for this package. Thus it makes the code harder to read. Based on bitter experience when I used to do it the other way.

A nice bonus is that it makes all the uses of `Electrometer.DetailedState` shorter by eliminating boilerplate.

**Review by r-owen — APPROVED** [review]
This looks like a useful addition. I have a few recommendations I hope you will consider before merging, but I will not block on them.

File: python/lsst/ts/electrometer/controller.py
Comment (wvreeven) [reply python/lsst/ts/electrometer/controller.py:248]: Done by me since Eric is on vacation.

File: python/lsst/ts/electrometer/controller.py
Comment (wvreeven) [reply python/lsst/ts/electrometer/controller.py:58]: Done by me since Eric is on vacation.

File: python/lsst/ts/electrometer/controller.py
Comment (wvreeven) [reply python/lsst/ts/electrometer/controller.py:263]: Done by me since Eric is on vacation.

File: python/lsst/ts/electrometer/controller.py
Comment (wvreeven) [reply python/lsst/ts/electrometer/controller.py:381]: I added the doc string and will leave it to Eric to eliminate `self.last_value` or not.

File: python/lsst/ts/electrometer/csc.py
Comment (wvreeven) [reply python/lsst/ts/electrometer/csc.py:265]: I'll leave this to Eric.

File: python/lsst/ts/electrometer/csc.py
Comment (wvreeven) [reply python/lsst/ts/electrometer/csc.py:54]: Done by me since Eric is on vacation.

**Pull request merged**
